danny-avila/LibreChat
# refactor(api): Central Logging 📜

## Summary

Introduces a central logging system to the backend (api). 

Log files are saved in `/api/logs`

Error logs are saved by default. Debug logs are enabled by default but can be turned off if not desired

More info: https://docs.librechat.ai/features/logging_system.html


- refactored the getAvailablePluginsController to cache the plugins since they are static, eliminating the need for the async addOpenAPISpecs call on every request. 
- fixed the Jest tests that failed due to recent changes in the logger configuration. Improved logging formatting has been implemented to avoid passing strings as the second argument, and sensitive values are now redacted for better security. - Boolean handling in the handleText function has been enhanced, and relevant tests have been written. 
- Circular dependencies introduced by accessing the logger have been resolved. 
- Unused tools have been removed, and overall logging statements have been refactored for clarity and efficiency. 
- The Redis initialization message now uses the 'info' level instead of 'debug'.

## Change Type


## Testing

During development, I've written unit tests to ensure that the boolean handling in handleText is functioning as expected. Additionally, after refactoring the logging, I conducted manual tests to confirm that logs are properly formatted and sensitive information is being redacted. For the caching of plugins, I verified that the application's performance improved due to the elimination of unnecessary repetitive calls. 

To reproduce my tests:
1. Run the updated Jest tests to check for the successful handling of booleans in text.
2. Inspect the logs to ensure that sensitive information is redacted and the new formatting is applied.
3. Monitor the performance of getAvailablePluginsController to confirm that caching is effective.

### **Test Configuration**:
- Environment variables set for DEBUG_LOGGING
- Updated `.env.example` with necessary configurations
- Logging configurations adjusted in `~/config/winston`

## Checklist
